src/simple_http.rs

```diff
@@ -133,18 +163,39 @@ impl SimpleHttpTransport {
             Err(_) => return Err(Error::HttpParseError),
         };
 
-        // Skip response header fields
-        while get_line(&mut reader, request_deadline)? != "\r\n" {}
+        // Parse response header fields
+        let mut content_length = None;
+        loop {
+            let line = get_line(&mut reader, request_deadline)?;
+
+            if line == "\r\n" {
+                break;
+            }
+
+            const CONTENT_LENGTH: &str = "content-length: ";
+            if line.to_lowercase().starts_with(CONTENT_LENGTH) {
+                content_length = Some(
+                    line[CONTENT_LENGTH.len()..]
+                        .trim()
+                        .parse::<usize>()
+                        .map_err(|_| Error::HttpParseError)?,
+                );
+            }
+        }
 
         if response_code == 401 {
             // There is no body in a 401 response, so don't try to read it
             return Err(Error::HttpErrorCode(response_code));
         }
 
+        let content_length = content_length.ok_or(Error::HttpParseError)?;
+
+        let mut buffer = vec![0; content_length];
```

Comment:
We definitely need to put a limit on this, but it's fine for now :). We'll do a followup PR to audit this because I think `get_line` may also be subject to memory exhaustion attacks.

Reply:
yes, that sounds prudent